tests/Doctrine/Tests/ODM/PHPCR/Functional/UnitOfWorkTest.php

```diff
@@ -185,4 +188,58 @@ public function testFetchingMultipleHierarchicalObjectsWithChildIdFirst()
         $this->assertSame($child->parent, $parent);
         $this->assertSame('parent', $parent->nodename);
     }
+
+    /**
+     * @expectedException Doctrine\ODM\PHPCR\Exception\OutOfBoundsException
+     * @expectedExceptionMessage Document "Doctrine\Tests\Models\CMS\CmsArticleFolder" does not allow children of type "Doctrine\Tests\Models\CMS\CmsGroup". Allowed child classes "Doctrine\Tests\Models\CMS\CmsArticle"
+     */
+    public function testRestrictChildrenInvalidChildren()
+    {
+        $articleFolder = new CmsArticleFolder();
+        $articleFolder->id = '/functional/articles';
+
+        $article = new CmsGroup();
+        $article->id = '/functional/articles/address';
+        $article->name = 'invalid-child';
+
+        $this->dm->persist($articleFolder);
+        $this->dm->persist($article);
```

Comment:
can you please add another test where the child would be persisted through cascading because its mapped on the parent document, instead of explicitly persist? i guess that also works, but we should make sure.